Clash between LITTLE_ENDIAN defines in RIOT and in arm-none-eabi-gcc 5.4

The newest `gcc-arm-none-eabi-5_4-2016q2-20160622` from [Launchpad](https://launchpad.net/gcc-arm-embedded/+download) creates a clash between LITTLE_ENDIAN defines in RIOT and in gcc. Here is the output from building the hello world example for the samr21-xpro:

```bash
In file included from $HOME/Git/RIOT/cpu/samd21/include/cpu_conf.h:23:0,
                 from $HOME/Git/RIOT/cpu/cortexm_common/include/cpu.h:35,
                 from $HOME/Git/RIOT/cpu/samd21/cpu.c:21:
$HOME/Git/RIOT/cpu/samd21/include/samd21.h:230:0: error: "LITTLE_ENDIAN" redefined [-Werror]
 #define LITTLE_ENDIAN          1
 ^
In file included from $HOME/Tools/gcc/gcc-arm-none-eabi-5_4-2016q2/arm-none-eabi/include/sys/types.h:67:0,
                 from $HOME/Tools/gcc/gcc-arm-none-eabi-5_4-2016q2/arm-none-eabi/include/stdio.h:48,
                 from $HOME/Git/RIOT/cpu/cortexm_common/include/cpu.h:33,
                 from $HOME/Git/RIOT/cpu/samd21/cpu.c:21:
$HOME/Tools/gcc/gcc-arm-none-eabi-5_4-2016q2/arm-none-eabi/include/machine/endian.h:17:0: note: this is the location of the previous definition
 #define LITTLE_ENDIAN _LITTLE_ENDIAN
 ^
```

And a [gist](https://gist.github.com/josephnoir/f6b7bb0491818aedd05e44473202efd1) with the complete output. Just in case the OS builds differ in some way, I'm using OS X.

--- a/cpu/samd21/include/samd21.h
+++ b/cpu/samd21/include/samd21.h
@@ -227,7 +227,9 @@ void I2S_Handler                 ( void );
  * \brief Configuration of the Cortex-M0+ Processor and Core Peripherals
  */
 
-#define LITTLE_ENDIAN          1
+#ifndef LITTLE_ENDIAN
+#   define LITTLE_ENDIAN       1
+#endif
 #define __CM0PLUS_REV          1         /*!< Core revision r0p1 */
 #define __MPU_PRESENT          0         /*!< MPU present or not */
 #define __NVIC_PRIO_BITS       2         /*!< Number of bits used for Priority Levels */

